### ISSUE TITLE
Training loss and accuracy not changing

### ISSUE THREAD

**Issue by ShastriRam** [fchollet/keras#6423]
I am running a CNN for left and right shoeprint classfication. I have 190,000 training images and I use 10% of it for validation. My model is setup as shown below. I get the paths of all the images, read them in and resize them. I normalize the image, and then fit it to the model. My issue is that I have stuck at a training accuracy of 62.5% and a loss of around 0.6615-0.6619. Is there something wrong that I am doing? How can I stop this from happening? 

from scipy import misc as image
import numpy as np
import os
import keras
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten
from keras.layers import Conv2D, MaxPooling2D
from keras.optimizers import SGD
from keras import backend as K
from keras.utils import np_utils

imageWidth = 50
imageHeight = 150

def get_filepaths(directory):
    file_paths = []  
    for filename in files:
            filepath = os.path.join(root, filename)
            file_paths.append(filepath)  # Add it to the list.
   return file_paths 

def cleanUpPaths(fullFilePaths): 
    cleanPaths = []
    for f in fullFilePaths:
        if f.endswith(".png"):
            cleanPaths.append(f)
    return cleanPaths
 
def getTrainData(paths):
    trainData = []
    for i in xrange(1,190000,2):
        im = image.imread(paths[i])
        im = image.imresize(im, (150,50))
        im = (im-255)/float(255)
        trainData.append(im)
    trainData = np.asarray(trainData)
    right = np.zeros(47500)
    left = np.ones(47500)
    trainLabels = np.concatenate((left, right))
    trainLabels = np_utils.to_categorical(trainLabels)   
    return (trainData, trainLabels)
   
#create the convnet
model = Sequential()

model.add(Conv2D(32, (3, 3), activation='relu', input_shape=(imageWidth,imageHeight,1),strides=1))#32
model.add(Conv2D(32, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), activation='relu',strides=1)) 
model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), activation='relu'))
model.add(MaxPooling2D(pool_size=(1, 3)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (1, 2), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 1)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(256, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(2, activation='softmax'))

sgd = SGD(lr=0.01)
model.compile(loss='categorical_crossentropy', optimizer='rmsprop',metrics=['accuracy'])

#*****prepare the training data******/

trainPaths = get_filepaths("better1/train")
trainPaths = cleanUpPaths(trainPaths)
(trainData, trainLabels) = getTrainData(trainPaths)
trainData = np.reshape(trainData,(95000,imageWidth,imageHeight,1)).astype('float32')
trainData = (trainData-255)/float(255)

#*****train the convnet********
model.fit(trainData, trainLabels, batch_size=500, epochs=50, validation_split=0.2)

#/****save the model and weights*****/
model.save('myConvnet_model5.h5');
model.save_weights('myConvnet_weights5.h5');

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by ShastriRam** [comment]
Just some interesting points to note:

1) I first tested this on 10 images I was having the same issue but changing the optimizer to adam and batch size to 4 worked.
2) I then tested on more and more images, but each time I would need to change the batch size to get improvements in the accuracy and loss. With 10,000 images I had to use a batch size of 500 and optimizer rmsprop. However, the accuracy and loss only really began to change after epoch 10.
3) I am now training on 190,000 images and I cannot increase the batch size as my GPU is at is max.

**Comment by rdelassus** [comment]
However, the accuracy and loss only really began to change after epoch 10.

A plateau at the beginning hints toward a bad initialization. Also this is more a stackoverflow question than an issue.

**Comment by ShastriRam** [comment]
I use the default initialization for all layers. I have been retraining with the 190,000 and even after 45 epochs with a batch size of 500, I still have  no change in accuracy and loss.

**Comment by engharat** [comment]
Agree that you should write to stackoverflow your question. Your network has way too much dropout, try removing all the dropout but the latest and see if the training loss converge towards zero.

**Comment by ShastriRam** [comment]
I apologize for posting here intially. I took your suggestions and posted to stackoverflow but i habvent gotten any responses. I have been working on this for more than a week and have no good results to show for it. Im going insane here and really desperate for some help. I removed the dropout layers but the last one, which is set at 0.5. Is this too much? I began retraining and the first few epochs have the same isse.

**Comment by engharat** [comment]
when you have some converging issue is always better to remove any
regularization to see if your training loss is correctly going to zero.
Then you will add regularization to fight overfitting. It is ok to do the
trial with only one dropout at 0.5, you could even remove it totally. let
see if the loss goes to zero after several epochs, if it will not be the
case then there is some errors somewhere

2017-04-28 10:11 GMT+02:00 TriniPhantom <notifications@github.com>:
